executor/runcexecutor/executor.go

```diff
@@ -524,14 +532,50 @@ func (p *procHandle) WaitForReady(ctx context.Context) error {
 func (p *procHandle) WaitForStart(ctx context.Context, startedCh <-chan int, started func()) error {
 	startedCtx, timeout := context.WithTimeout(ctx, 10*time.Second)
 	defer timeout()
-	var err error
 	select {
 	case <-startedCtx.Done():
 		return errors.New("runc started message never received")
-	case pid, ok := <-startedCh:
+	case runcPid, ok := <-startedCh:
 		if !ok {
 			return errors.New("runc process failed to send pid")
 		}
+
+		// we need the process for the runc monitor process so we can send
+		// SIGWINCH signals to it, if we send them to the process inside the
+		// container nothing happens.
+		var err error
+		p.MonitorProcess, err = os.FindProcess(runcPid)
+		if err != nil {
+			return errors.Wrapf(err, "unable to find runc monitor process for pid %d", runcPid)
+		}
+
+		// the pid reported from the started channel seems to be the pid
+		// of the runc monitor process, but we will need to send signals
+		// to the process running in the container so load the process pid
+		// from the pidfile.  Unfortunately the pidfile will not be written
+		// by the time we get the runc pid via the channel, so we need to retry.
+		var pid int
+		for {
+			pidData, err := os.ReadFile(p.pidfile)
+			if err != nil {
+				return errors.Wrap(err, "unable to read pidfile after runc process started")
+			}
+			pid, err = strconv.Atoi(string(pidData))
+			if err != nil {
+				select {
+				case <-ctx.Done():
+					if len(pidData) == 0 {
+						return errors.Errorf("context cancelled before runc wrote to pidfile")
+					}
+					return errors.Wrap(err, "context cancelled before we found valid runc pid")
+				case <-time.After(50 * time.Millisecond):
```

Comment:
not sure what value to use here for retry, in my experience it was taking ~30-40ms for the pidfile to be updated.  We can shorten this duration, not sure there is much harm in making this loop faster, maybe 5-10ms?